### ISSUE TITLE
Sync and Async methods should have same order of parameters and proper return type

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#660]
In case there is a synchronous method that has an asynchronous counterpart, both methods should have the same kind of parameters, in the same order.

The return value of the asynchronous method should be `Task` for `void` synchronous methods or `Task<Xyz>` for synchronous methods that return a `Xyz`.

Example:
```c#
int DoSomething(int x, int y, int z);
```
should have following async counterpart (if needed at all)
```c#
Task<int> DoSomethingAsync(int x, int y, int z);
```

In case the counterpart is e.g.
```c#
Task<int> DoSomethingAsync(int x, int z, int y);
```
we should report a violation because the parameters `y` and `z` are switched.
